lib/autoupdate/agent/updater.go

```diff
@@ -0,0 +1,338 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package agent
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/fs"
+	"log/slog"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/renameio/v2"
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v3"
+
+	"github.com/gravitational/teleport/api/client/webclient"
+	libdefaults "github.com/gravitational/teleport/lib/defaults"
+	libutils "github.com/gravitational/teleport/lib/utils"
+)
+
+var (
+	featureFlag int
```

Comment:
We would probably need to move this one to `lib/autoupdate` level, since both agent and tools packages use this variable.
Also `updateVersionEnvVar = "TELEPORT_UPDATE_VERSION"` kind of common
Can do it in mine PR after merging this one

Reply:
That sounds good for now. Eventually I think the agent updater should query the cluster for this.